### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by jbygdell**
Changes made:

* Added option to generate certificates through cert-manger
* Added option to add ingress annotations
* Switched to newer `ingressClassName`, for ingress selection
* Reworked the test suite to test the deployment functionality, i.e. external facing services responds properly
* Removed all Python based helper files, as these tested the ingestion pipeline and that is not meant to be done through the helm tests
* Updated the README files
* Bumped all chart versions

### CODE REVIEW

File: charts/sda-svc/templates/backup-deploy.yaml [hunk 1]
```diff
@@ -242,19 +240,12 @@ spec:
...
-                name: {{ .Values.global.tls.caCert.secretName }}
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
+                name: {{ template "sda.fullname" . }}-backup-certs
             {{- else }}
-            - secret:
-                name: {{ template "sda.fullname" . }}-caCert
+              - secret:
+                  name: {{ required "An certificate issuer or a TLS secret name is requrired for backup" .Values.backup.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/backup-deploy.yaml:248]:
```suggestion
                  name: {{ required "An certificate issuer or a TLS secret name is required for backup" .Values.backup.tls.secretName }}
```

File: charts/sda-svc/templates/auth-deploy.yaml [hunk 2]
```diff
@@ -138,40 +146,35 @@ spec:
...
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
-            {{- else }}
+                name: {{ template "sda.fullname" . }}-auth-certs
+          {{- else }}
             - secret:
-                name: {{ template "sda.fullname" . }}-caCert
-            {{- end }}
+                name: {{ required "An certificate issuer or a TLS secret name is requrired for auth" .Values.auth.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/auth-deploy.yaml:176]:
```suggestion
                name: {{ required "An certificate issuer or a TLS secret name is required for auth" .Values.auth.tls.secretName }}
```

File: charts/sda-svc/templates/doa-deploy.yaml [hunk 3]
```diff
@@ -269,18 +270,13 @@ spec:
...
             - secret:
-                name: {{ .Values.global.tls.caCert.secretName }}
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
-            {{- else }}
-            - secret:
-                name: {{ template "sda.fullname" . }}-cacert
-            {{- end }}
+                name: {{ required "An certificate issuer or a TLS secret name is requrired for doa" .Values.doa.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/doa-deploy.yaml:278]:
```suggestion
                name: {{ required "An certificate issuer or a TLS secret name is required for doa" .Values.doa.tls.secretName }}
```

File: charts/sda-svc/templates/download-deploy.yaml [hunk 4]
```diff
@@ -206,19 +217,12 @@ spec:
...
-                name: {{ .Values.global.tls.caCert.secretName }}
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
+                name: {{ template "sda.fullname" . }}-download-certs
             {{- else }}
-            - secret:
-                name: {{ template "sda.fullname" . }}-cacert
+              - secret:
+                  name: {{ required "An certificate issuer or a TLS secret name is requrired for download" .Values.download.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/download-deploy.yaml:225]:
```suggestion
                  name: {{ required "An certificate issuer or a TLS secret name is required for download" .Values.download.tls.secretName }}
```

File: charts/sda-svc/templates/finalize-deploy.yaml [hunk 5]
```diff
@@ -159,19 +157,12 @@ spec:
...
             - secret:
-                name: {{ .Values.global.tls.caCert.secretName }}
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
+                name: {{ template "sda.fullname" . }}-finalize-certs
             {{- else }}
             - secret:
-                name: {{ template "sda.fullname" . }}-cacert
+                name: {{ required "An certificate issuer or a TLS secret name is requrired for finalize" .Values.finalize.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/finalize-deploy.yaml:165]:
```suggestion
                name: {{ required "An certificate issuer or a TLS secret name is required for finalize" .Values.finalize.tls.secretName }}
```

File: charts/sda-svc/templates/ingest-deploy.yaml [hunk 6]
```diff
@@ -258,19 +256,12 @@ spec:
...
-                name: {{ .Values.global.tls.caCert.secretName }}
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
+                name: {{ template "sda.fullname" . }}-ingest-certs
             {{- else }}
-            - secret:
-                name: {{ template "sda.fullname" . }}-cacert
+              - secret:
+                  name: {{ required "An certificate issuer or a TLS secret name is requrired for ingest" .Values.ingest.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/ingest-deploy.yaml:264]:
```suggestion
                  name: {{ required "An certificate issuer or a TLS secret name is required for ingest" .Values.ingest.tls.secretName }}
```

File: charts/sda-svc/templates/intercept-deploy.yaml [hunk 7]
```diff
@@ -110,19 +110,12 @@ spec:
...
-                name: {{ .Values.global.tls.caCert.secretName }}
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
+                name: {{ template "sda.fullname" . }}-intercept-certs
             {{- else }}
-            - secret:
-                name: {{ template "sda.fullname" . }}-cacert
+              - secret:
+                  name: {{ required "An certificate issuer or a TLS secret name is requrired for ingest" .Values.ingest.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/intercept-deploy.yaml:118]:
```suggestion
                  name: {{ required "An certificate issuer or a TLS secret name is required for ingest" .Values.ingest.tls.secretName }}
```

File: charts/sda-svc/templates/mapper-deploy.yaml [hunk 8]
```diff
@@ -157,20 +155,13 @@ spec:
...
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
-            {{- else }}
+                name: {{ template "sda.fullname" . }}-mapper-certs
+          {{- else if .Values.mapper.tls.secretName }}
             - secret:
-                name: {{ template "sda.fullname" . }}-cacert
-            {{- end }}
+                name: {{ required "An certificate issuer or a TLS secret name is requrired for mapper" .Values.mapper.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/mapper-deploy.yaml:163]:
```suggestion
                name: {{ required "An certificate issuer or a TLS secret name is required for mapper" .Values.mapper.tls.secretName }}
```

File: charts/sda-svc/templates/release-test-deploy.yml [hunk 9]
```diff
@@ -196,65 +94,56 @@ spec:
...
-              - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                path: ca.crt
+          {{- if or .Values.global.tls.clusterIssuer .Values.global.tls.issuer }}
+            - secret:
+                name: {{ template "sda.fullname" . }}-tester-certs
           {{- else }}
-          - secret:
-              name: {{ template "sda.fullname" . }}-caCert
+            - secret:
+                name: {{ required "An certificate issuer or a TLS secret name is requrired for releasetest" .Values.releasetest.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/release-test-deploy.yml:102]:
```suggestion
                name: {{ required "An certificate issuer or a TLS secret name is required for releasetest" .Values.releasetest.tls.secretName }}
```

File: charts/sda-svc/templates/s3-inbox-deploy.yaml [hunk 10]
```diff
@@ -121,13 +121,11 @@ spec:
...
-        {{- if or .Values.global.auth.jwtPub .Values.global.elixir.pubKey }}
+        {{- if.Values.global.auth.jwtPub }}
         - name: SERVER_JWTPUBKEYPATH
           value: {{ include "jwtPath" . }}
-        {{- end }}
-        {{- if and .Values.global.elixir.oidcdHost .Values.global.elixir.jwkPath }}
-        - name: SERVER_JWTPUBKEYURL
-          value: "{{ .Values.global.elixir.oidcdHost }}{{ .Values.global.elixir.jwkPath }}"
+        {{- else }}
+          {{ fail "The global.auth.jwtPub is requrired for s3Inbox" }}
```
Comment (norling) [inline charts/sda-svc/templates/s3-inbox-deploy.yaml:128]:
```suggestion
          {{ fail "The global.auth.jwtPub is required for s3Inbox" }}
```

File: charts/sda-svc/templates/s3-inbox-deploy.yaml [hunk 11]
```diff
@@ -159,32 +157,29 @@ spec:
...
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
-            {{- else }}
-            - secret:
-                name: {{ template "sda.fullname" . }}-cacert
-            {{- end }}
+          {{- if or .Values.global.tls.clusterIssuer .Values.global.tls.issuer }}
+                name: {{ template "sda.fullname" . }}-inbox-certs
+          {{- else }}
+                name: {{ required "An certificate issuer or a TLS secret name is requrired for S3Inbox" .Values.s3Inbox.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/s3-inbox-deploy.yaml:171]:
```suggestion
                name: {{ required "An certificate issuer or a TLS secret name is required for S3Inbox" .Values.s3Inbox.tls.secretName }}
```

File: charts/sda-svc/templates/sftp-inbox-deploy.yaml [hunk 12]
```diff
@@ -167,17 +167,12 @@ spec:
...
+            {{- else}}
             - secret:
-                name: {{ .Values.global.tls.caCert.secretName }}
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
-            {{- else }}
-            - secret:
-                name: {{ template "sda.fullname" . }}-cacert
+                name: {{ required "An certificate issuer or a TLS secret name is requrired for sftpInbobx" .Values.sftpInbox.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/sftp-inbox-deploy.yaml:175]:
```suggestion
                name: {{ required "An certificate issuer or a TLS secret name is required for sftpInbobx" .Values.sftpInbox.tls.secretName }}
```

File: charts/sda-svc/templates/verify-deploy.yaml [hunk 13]
```diff
@@ -211,19 +211,12 @@ spec:
...
-                name: {{ .Values.global.tls.caCert.secretName }}
-                items:
-                - key: {{ required "A name for the CA cert is required" .Values.global.tls.caCert.secretKey }}
-                  path: ca.crt
-            {{- else }}
+                name: {{ template "sda.fullname" . }}-verify-certs
+            {{- else if .Values.verify.tls.secretName }}
             - secret:
-                name: {{ template "sda.fullname" . }}-cacert
+                name: {{ required "An certificate issuer or a TLS secret name is requrired for verify" .Values.verify.tls.secretName }}
```
Comment (norling) [inline charts/sda-svc/templates/verify-deploy.yaml:219]:
```suggestion
                name: {{ required "An certificate issuer or a TLS secret name is required for verify" .Values.verify.tls.secretName }}
```

File: dev_tools/scripts/ssl.cnf [hunk 14]
```diff
@@ -182,7 +182,7 @@ subjectKeyIdentifier = hash
 authorityKeyIdentifier = keyid,issuer:always
 keyUsage = critical, digitalSignature, keyEncipherment
 extendedKeyUsage = clientAuth, serverAuth
-subjectAltName = DNS:inbox,DNS:inbox.default.svc.cluster.local,DNS:inbox.default
+subjectAltName = DNS:sda-sda-svc-inbox,DNS:inbox.default.svc.cluster.local,DNS:inbox.default,DNS:loaclhost
```
Comment (norling) [inline dev_tools/scripts/ssl.cnf:185]:
```suggestion
subjectAltName = DNS:sda-sda-svc-inbox,DNS:inbox.default.svc.cluster.local,DNS:inbox.default,DNS:localhost
```

File: dev_tools/scripts/ssl.cnf [hunk 15]
```diff
@@ -191,7 +191,7 @@ subjectKeyIdentifier = hash
 authorityKeyIdentifier = keyid,issuer:always
 keyUsage = critical, digitalSignature, keyEncipherment
 extendedKeyUsage = clientAuth, serverAuth
-subjectAltName = DNS:doa,DNS:doa.default.svc.cluster.local,DNS:doa.default
+subjectAltName = DNS:sda-sda-svc-doa,DNS:doa.default.svc.cluster.local,DNS:doa.default,DNS:loaclhost
```
Comment (norling) [inline dev_tools/scripts/ssl.cnf:194]:
```suggestion
subjectAltName = DNS:sda-sda-svc-doa,DNS:doa.default.svc.cluster.local,DNS:doa.default,DNS:localhost
```

File: dev_tools/scripts/ssl.cnf [hunk 16]
```diff
@@ -200,7 +200,7 @@ subjectKeyIdentifier = hash
 authorityKeyIdentifier = keyid,issuer:always
 keyUsage = critical, digitalSignature, keyEncipherment
 extendedKeyUsage = clientAuth, serverAuth
-subjectAltName = DNS:download,DNS:download.default.svc.cluster.local,DNS:download.default
+subjectAltName = DNS:sda-sda-svc-download,DNS:download.default.svc.cluster.local,DNS:download.default,DNS:loaclhost
```
Comment (norling) [inline dev_tools/scripts/ssl.cnf:203]:
```suggestion
subjectAltName = DNS:sda-sda-svc-download,DNS:download.default.svc.cluster.local,DNS:download.default,DNS:localhost
```

File: dev_tools/scripts/ssl.cnf [hunk 17]
```diff
@@ -209,7 +209,7 @@ subjectKeyIdentifier = hash
 authorityKeyIdentifier = keyid,issuer:always
 keyUsage = critical, digitalSignature, keyEncipherment
 extendedKeyUsage = clientAuth, serverAuth
-subjectAltName = DNS:auth,DNS:auth.default.svc.cluster.local,DNS:auth.default
+subjectAltName = DNS:sda-sda-svc-auth,DNS:auth.default.svc.cluster.local,DNS:auth.default,DNS:loaclhost
```
Comment (norling) [inline dev_tools/scripts/ssl.cnf:212]:
```suggestion
subjectAltName = DNS:sda-sda-svc-auth,DNS:auth.default.svc.cluster.local,DNS:auth.default,DNS:localhost
```

**Review by norling — APPROVED** [review]
Some typos, but otherwise: looks sound as far as I can tell! :)

**Review by blankdots — COMMENTED** [review]
there is a small bug dev_tools/config/s3.yaml where inbox.s3Bucket is "archive"

**Review by blankdots — APPROVED** [review]
LGTM

**Comment by jbygdell** [comment]
> there is a small bug dev_tools/config/s3.yaml where inbox.s3Bucket is "archive"

Apparently that issue is really old, but I fixed it here.

**Review by blankdots — APPROVED** [review]

**Pull request merged**